### PR TITLE
CLOSED Incremental migration path from QueryParams to QuerySpec #347

### PR DESCRIPTION
**Pull request by Ramblurr**
see #347

### CODE REVIEW

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    private ResourceRegistry resourceRegistry;
+
+    public DefaultQueryParamsConverter(ResourceRegistry resourceRegistry) {
+        this.resourceRegistry = resourceRegistry;
+    }
+
+    @Override
+    public QueryParams fromParams(Class<?> rootType, QuerySpec querySpec) {
+        QueryParams params = new QueryParams();
```
Comment (remmeier) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java]: you may want to call getRelatedSpecs as well and serialize those QuerySpecs as well (to process parameters other than for the root type)

**Comment by remmeier** [comment]
looks good, potentially you can also converted nested QuerySpecs for parameters other than the root-type.

**Comment by Ramblurr** [comment]
thanks @remmeier. I'm not sure how the nested QuerySpecs work. 

Could you give a short example or a test that would fail with my current converter?

**Comment by Ramblurr** [comment]
I just added one more commit that converts from QuerySpec to QueryParams by default.

**Comment by remmeier** [comment]
documentation should cover the getRelatedSpec.

**Comment by Ramblurr** [comment]
Refactored this against master post-3.0 and opened a new PR #356

**Pull request closed without merging**